cellgrid/capped_distance.py

```diff
@@ -91,11 +93,24 @@ def _calculate_distance_array_size(cg1, cg2):
     return N
 
 
-def capped_self_distance_array(a, max_dist, box):
-    cga = CellGrid(box, max_dist, a)
+def capped_self_distance_array(a, cellsize, box, particles_per_cell = 30):
+    """
+    Optimized to identify pair contacts within a distance
+    Cellsize is changed based on the maximum of number of 
+    particles per cell and provided cellsize
+    """
+    if particles_per_cell is not None:
+        cellsize = min(max(cellsize, _per_cell(a.shape[0], box, particles_per_cell)), 0.33*min(box))
```

Comment:
Where does the limit of 0.33 * min(box) come from?  Is this to make sure the minimum dimensions are 3 x 3 x 3?